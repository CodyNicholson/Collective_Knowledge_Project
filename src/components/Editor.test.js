//This file was begun using code from Link.react.test.js, from the Jest Testing React Apps Guide
//https://facebook.github.io/jest/docs/en/tutorial-react.html
import React from 'react';
import { Editor } from './Editor';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

var paramsData= {slug:null};
var matchData= {params: paramsData};
var count = 0;
function onLoadF() {count++;};

test('Tags module test', () => {
  const component = renderer.create(
    <MemoryRouter>
    	<Editor match={matchData} onLoad={onLoadF}>
    	</Editor>
    </MemoryRouter>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  expect(count).toBe(1);
});