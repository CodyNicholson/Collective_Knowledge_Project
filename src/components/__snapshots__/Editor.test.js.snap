// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags module test 1`] = `
<div
  className="editor-page"
>
  <div
    className="container page"
  >
    <div
      className="row"
    >
      <div
        className="col-md-10 offset-md-1 col-xs-12"
      >
        <form>
          <fieldset>
            <fieldset
              className="form-group"
            >
              <input
                className="form-control form-control-lg"
                onChange={[Function]}
                placeholder="Article Title"
                type="text"
                value={undefined}
              />
            </fieldset>
            <fieldset
              className="form-group"
            >
              <input
                className="form-control"
                onChange={[Function]}
                placeholder="What's this article about?"
                type="text"
                value={undefined}
              />
            </fieldset>
            <fieldset
              className="form-group"
            >
              <textarea
                className="form-control"
                onChange={[Function]}
                placeholder="Write your article (in markdown)"
                rows="8"
                value={undefined}
              />
            </fieldset>
            <fieldset
              className="form-group"
            >
              <input
                className="form-control"
                onChange={[Function]}
                onKeyUp={[Function]}
                placeholder="Enter tags"
                type="text"
                value={undefined}
              />
              <div
                className="tag-list"
              />
            </fieldset>
            <button
              className="btn btn-lg pull-xs-right btn-primary"
              disabled={undefined}
              onClick={[Function]}
              type="button"
            >
              Publish Article
            </button>
          </fieldset>
        </form>
      </div>
    </div>
  </div>
</div>
`;
